src/backend/base/langflow/initial_setup/setup.py

```diff
@@ -543,52 +550,96 @@ async def load_flows_from_directory() -> None:
-        _flows_path = Path(flows_path)
-        files = [f for f in _flows_path.iterdir() if f.is_file()]
-        for file_path in files:
-            if file_path.suffix != ".json":
+        for file_path in Path(flows_path).iterdir():
+            if not file_path.is_file() or file_path.suffix != ".json":
                 continue
             logger.info(f"Loading flow from file: {file_path.name}")
             async with async_open(file_path, "r", encoding="utf-8") as f:
                 content = await f.read()
-            flow = orjson.loads(content)
-            no_json_name = file_path.stem
-            flow_endpoint_name = flow.get("endpoint_name")
-            if _is_valid_uuid(no_json_name):
-                flow["id"] = no_json_name
-            flow_id = flow.get("id")
-
-            existing = await find_existing_flow(session, flow_id, flow_endpoint_name)
-            if existing:
-                logger.debug(f"Found existing flow: {existing.name}")
-                logger.info(f"Updating existing flow: {flow_id} with endpoint name {flow_endpoint_name}")
-                for key, value in flow.items():
-                    if hasattr(existing, key):
-                        # flow dict from json and db representation are not 100% the same
-                        setattr(existing, key, value)
-                existing.updated_at = datetime.now(tz=timezone.utc).astimezone()
-                existing.user_id = user_id
-
-                # Generally, folder_id should not be None, but we must check this due to the previous
-                # behavior where flows could be added and folder_id was None, orphaning
-                # them within Langflow.
-                if existing.folder_id is None:
-                    folder_id = await get_default_folder_id(session, user_id)
-                    existing.folder_id = folder_id
-
-                session.add(existing)
-            else:
-                logger.info(f"Creating new flow: {flow_id} with endpoint name {flow_endpoint_name}")
+            await upsert_flow_from_file(content, file_path, session, user.id)
+
+
+async def load_bundles_from_urls() -> tuple[list[tempfile.TemporaryDirectory], list[str]]:
+    component_paths = []
+    temp_dirs = []
+    settings_service = get_settings_service()
+    bundle_urls = settings_service.settings.bundle_urls
+    if not bundle_urls:
+        return [], []
+    if not settings_service.auth_settings.AUTO_LOGIN:
+        logger.warning("AUTO_LOGIN is disabled, not loading flows from URLs")
+
+    async with async_session_scope() as session:
+        user = await get_user_by_username(session, settings_service.auth_settings.SUPERUSER)
+        if user is None:
+            msg = "Superuser not found in the database"
+            raise NoResultFound(msg)
+        user_id = user.id
+
+        for url in bundle_urls:
+            async with httpx.AsyncClient(follow_redirects=True) as client:
+                response = await client.get(url)
```

Comment:
One QoL I'd add is the possibility to just pass a github repo url and we take care of building the url for the zip.

Reply:
We could have detection of GH repo URLs as follow-up work ?

Reply:
Sounds good.

Reply:
Can the URL be easily found somewhere?

Reply:
Here's the GitHub doc for the URL format: https://docs.github.com/en/repositories/working-with-files/using-files/downloading-source-code-archives#source-code-archive-urls